### PR TITLE
feat: #528 Add-Key to a role

### LINKED ISSUE #528: Support Adding External Keys
The CLI currently has a `gen-key` method and a `revoke-key` method, but not an `add-key` method. This makes it hard for users with external keys, for example KMS or otherwised managed key services, to add their keys to the `root.json`. Whilst it can be done with json manipulation, it would be much nicer if the tuf cli supported it natively to avoid mistakes or misconfiguration.

I propose the following

```
tuff add-key <role>
--keytype # eg. "rsa"
--public-value # The value of the public key
--scheme #eg "rsassa-pss-sha256"

# Returns the key-id
```

This will then add the key to the root,json in both the `keys`, and the roles fields with the correct hashed key id, and will return a key id which can then be used in subsequent signing requests.

### PR DESCRIPTION
**Pull request by ChevronTango**
Please fill in the fields below to submit a pull request.  The more information that is provided, the better.

Fixes #528 

Release Notes: Introduces the `add-key` command

**Types of changes**:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). **Please ensure that your PR title** is a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) breaking change (with a `!`, as in `feat!: change foo`). 

**Description of the changes being introduced by the pull request**:
Introduces the `add-key` command

**Please verify and check that the pull request fulfills the following requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: cmd/tuf/add_key.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+`)
+}
+
+func cmdAddKey(args *docopt.Args, repo *tuf.Repo) error {
+	role := args.String["<role>"]
+	var keyids []string
+
+	keyScheme := data.KeySchemeEd25519
+	switch t := args.String["--scheme"]; t {
+	case string(data.KeySchemeEd25519),
```
Comment (kommendorkapten) [inline cmd/tuf/add_key.go:35]: I think we should not default to ed25519 here, I think it's better UX to abort with a nice error message like `key type %s is not supported` (where the variable `t` should be used during logging).

File: cmd/tuf/add_key.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		}
+	} else {
+		expires = data.DefaultExpires(role)
+	}
+	keyids, err = repo.AddKeyWithSchemeAndExpires(role, expires, keyScheme, publicValue)
+	if err != nil {
+		return err
+	}
+	for _, id := range keyids {
+		fmt.Fprintf(os.Stdout, "Add key with ID %s", role, keyScheme, id)
```
Comment (kommendorkapten) [inline cmd/tuf/add_key.go:73]: You are missing the last two arguments here in the format string (can be seen by running `go vet`):
```shell
$  go vet ./...
# github.com/theupdateframework/go-tuf/cmd/tuf
cmd/tuf/add_key.go:73:3: fmt.Fprintf call needs 1 arg but has 3 args
```

File: README.md [hunk 3]
```diff
@@ -49,6 +49,15 @@ snapshots (i.e. by passing `--consistent-snapshot=false`). If consistent
 snapshots should be generated, the repository will be implicitly
 initialized to do so when generating keys.
 
+#### `tuf add-key [--scheme=<scheme>] [--expires=<days>] [--public-value=<path>] <role>`
```
Comment (kommendorkapten) [inline README.md:52]: NIT: maybe call the parameter `--public-key` instead of `--public-value`?

**Comment by kommendorkapten** [comment]
Nice work 🚀  
Left some comments that should be fixed.

**Comment by ChevronTango** [comment]
> Nice work 🚀 Left some comments that should be fixed.

Brilliant! Thanks for those great spots! Just patched it now so should be ready to go! :)

**Comment by kommendorkapten** [comment]
Sorry for not responding to this, been very busy. I will go over your latest changes next week 👍

**Review by kommendorkapten — APPROVED** [review]

**Review by rdimitrov — APPROVED** [review]
Thanks, @ChevronTango! 🚀

**Pull request merged**